pagai/services/database_explorer.py

```diff
@@ -81,36 +112,50 @@ def get_column(self, column: str, table: str, schema: str = None):
             # as case insensitive).
             return table.c[column.lower()]
 
-    def get_table(self, table: str, schema=None, limit=1000, filters=[]):
+    def get_table_rows(
+        self, table_name: str, columns_names: List[str], schema=None, limit=100, filters=[]
+    ):
         """
         Return content of a table with a limit
         """
-        table = self._get_sql_table(table, schema)
+        table = self.get_sql_alchemy_table(table_name, schema)
         select = table.select()
 
         # Add filtering if any
         for filter_ in filters:
-            col = self.get_column(
+            col = self.get_sql_alchemy_column(
                 filter_["sqlColumn"]["column"], filter_["sqlColumn"]["table"], schema
             )
             rel_method = SQL_RELATIONS_TO_METHOD[filter_["relation"]]
             select = select.where(getattr(col, rel_method)(filter_["value"]))
 
         # Return as JSON serializable object
-        return [[col for col in row] for row in self._sql_engine.execute(select.limit(limit))]
+        return [
+            [get_col_from_row_result(row, col) for col in columns_names]
+            for row in self._sql_engine.execute(select.limit(limit))
+        ]
 
-    def explore(self, table: str, limit, schema=None, filters=[]):
+    def explore(self, table_name: str, schema: str, limit: int, filters=[]):
         """
         Returns the first rows of a table alongside the column names.
         """
+        self.check_connection_exists()
+
         try:
+            columns_names = self.get_column_names(table_name, schema=schema)
```

Comment:
why is this a variable and not the result of `self.get_table_rows(...)` ?

Reply:
I changed a bit how things work here

Reply:
better :+1: